StoredProcedureEFCore/OutputParameter.cs

```diff
@@ -3,17 +3,31 @@
 
 namespace StoredProcedureEFCore
 {
-  internal class OutputParam<T> : IOutParam<T>
-  {
-    public OutputParam(DbParameter param)
+    internal class OutputParam<T> : IOutParam<T>
     {
-      _dbParam = param;
-    }
+        public OutputParam(DbParameter param)
+        {
+            _dbParam = param;
+        }
+
+        public T Value
+        {
+            get
+            {
+                if (_dbParam.Value is System.DBNull)
+                {
+                    if (Nullable.GetUnderlyingType(typeof(T)) != null)
+                    {
+                        return default(T);
+                    }
+                }
 
-    public T Value => (T)Convert.ChangeType(_dbParam.Value, typeof(T));
+                return (T)Convert.ChangeType(_dbParam.Value, typeof(T));
```

Comment:
I'll have to check why I've used a ChangeType here. A cast should probably be enough.

Reply:
I have tested it. A cast ist not sufficient, for example dbo.ReturnBoolean (Program.cs line 39) would throw an exception as it is not possible to cast the int (bit on sql server) value to a boolean value.